eng/Packages/TestOnly-net6.0.props

```diff
@@ -3,6 +3,9 @@
   <ItemGroup Condition="'$(TargetFramework)' == 'net6.0'">
     <PackageVersion Include="Microsoft.AspNetCore.Mvc.Testing" Version="6.0.1" />
     <PackageVersion Include="Microsoft.AspNetCore.TestHost" Version="6.0.1" />
+    <PackageVersion Include="Microsoft.Extensions.Diagnostics.HealthChecks" Version="$(MicrosoftExtensionsDiagnosticsHealthChecksVersion)" />
+    <PackageVersion Include="Microsoft.Extensions.Http" Version="6.0.0" />
```

Comment:
Shouldn't all of these versions come from versions.props using dependency flow? Also, why are we pinning down the dependency to 6.0 version, we should be always using latest version of the extension packages.

Reply:
I think once Eric's compatibility PR goes in, we can cleanup the files in eng/Packages to only ever depend on the .NET 8 extensions. At that point, we can also cleanup `#if` throughout the code base to remove dead branches.